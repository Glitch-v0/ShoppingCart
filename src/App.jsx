/* eslint-disable no-console */
import { Outlet } from 'react-router-dom'
import { useState, useEffect } from 'react'
import NavBar from './NavBar'

function App() {
  const [cartItems, updateCart] = useState([])
  const [products, updateProducts] = useState([])

  useEffect(() => {
    fetch('https://fakestoreapi.com/products?limit=5', { mode: 'cors' })
      .then((response) => {
        if (response.status >= 400) {
          throw new Error('server error')
        }
        return response.json()
      })
      .then((response) => updateProducts(response))
      .catch((error) => console.error('Fetch error:', error))
  }, [])
  const totalItemsInCart = cartItems.reduce(
    (total, item) => total + item.quantity,
    0,
  )
  return (
    <div>
      <NavBar context={{ totalItemsInCart }} />
      <main>
        <Outlet
          context={{ products, cartItems, updateCart, totalItemsInCart }}
        />
      </main>
    </div>
  )
}

export default App
