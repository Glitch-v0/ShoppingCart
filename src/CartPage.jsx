import { useOutletContext } from 'react-router-dom'
import CardItemCard from './CartItemCard'

function CartPage() {
  const { cartItems } = useOutletContext([])
  const { totalItemsInCart } = useOutletContext([])
  return (
    <div>
      <h1>Your Shopping Cart:</h1>
      <h2>You currently have {totalItemsInCart} items in your cart.</h2>
      {cartItems.map((cartItem) => (
        <CardItemCard item={cartItem} key={cartItem.id * 10} />
      ))}
    </div>
  )
}

export default CartPage
