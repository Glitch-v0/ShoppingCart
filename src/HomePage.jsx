import fancyImage from './assets/fancy.jpg'

function HomePage() {
  return (
    <div className="homepage-container">
      <h1 className="welcome">Fashion Boutique</h1>
      <img
        src={fancyImage}
        alt="A lady carrying several bags of merchandise"
        height="200px"
        className="welcome-image"
      />
      <h2>Top Quality, Classy Organics</h2>
      <br />
      <h3>
        Finally, a place where you feel comfortable spending your money. Worry
        no longer about that empty feeling in your wallet. Instead, maximize the
        profit of our shareholders and fulfill our wildest dreams.
      </h3>
    </div>
  )
}

export default HomePage
