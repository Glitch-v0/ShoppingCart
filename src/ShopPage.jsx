import { useOutletContext } from 'react-router-dom'
import ItemCard from './ItemCard'

function ShopPage() {
  const { products } = useOutletContext([])
  return (
    <div className="shopping-container">
      <h1>Shopping Page</h1>
      <div className="page-body">
        {products.map((product) => (
          <ItemCard item={product} key={product.id} />
        ))}
      </div>
    </div>
  )
}

export default ShopPage
