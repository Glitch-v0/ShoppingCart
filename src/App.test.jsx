import React from 'react'
import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { BrowserRouter } from 'react-router-dom'
import App from './App'
import HomePage from './HomePage'
import ShopPage from './ShopPage'
import CartPage from './CartPage'

describe('App Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('Home Page', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<HomePage />)
    expect(asFragment()).toMatchSnapshot()
  })
})
// Fix the useOutletContext error
// describe('Shop Page', () => {
//   it('matches the snapshot', () => {
//     const { asFragment } = render(<ShopPage />)
//     expect(asFragment()).toMatchSnapshot()
//   })
// })
