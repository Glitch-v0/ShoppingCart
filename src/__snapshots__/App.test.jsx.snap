// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="nav-container"
    >
      <nav>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/shop"
        >
          Shop
        </a>
        <a
          href="/cart"
        >
          Cart: 0 items
        </a>
      </nav>
    </div>
    <main />
  </div>
</DocumentFragment>
`;

exports[`Home Page > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="homepage-container"
  >
    <h1
      class="welcome"
    >
      Fashion Boutique
    </h1>
    <img
      alt="A lady carrying several bags of merchandise"
      class="welcome-image"
      height="200px"
      src="/src/assets/fancy.jpg"
    />
    <h2>
      Top Quality, Classy Organics
    </h2>
  </div>
</DocumentFragment>
`;
